src/test/java/org/nuxeo/ecm/platform/pdf/tests/TestUtils.java

```diff
@@ -87,4 +82,42 @@ static String calculateMd5(File inFile) throws IOException {
         return md5;
     }
 
+    static boolean hasTextOnAllPages(Blob blob, String watermark) {
+        try (PDDocument doc = PDDocument.load(blob.getStream())) {
+            for (int i = 1; i <= doc.getNumberOfPages(); i++) {
+                if (!TestUtils.extractText(doc, i, i).replace("\n", "").contains(watermark)) {
+                    return false;
+                }
+            }
+            return true;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Remove the `printStackTrace` and add a comment explaining when this exception occurs. If unknown and not happening during tests, then just re-raise as a `NuxeoException`.